compiler/rustc_borrowck/src/type_check/mod.rs

```diff
@@ -2242,6 +2242,23 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                             }
                         }
                     }
+                    CastKind::Transmute => {
+                        let ty_from = op.ty(body, tcx);
+                        let src_layout = self.tcx().layout_of(self.param_env.and(ty_from));
+                        let dst_layout = self.tcx().layout_of(self.param_env.and(*ty));
```

Comment:
This can cause cycle errors if the casted type contains generators. Maybe we should just skip such a check at this stage.

Reply:
I was definitely unsure of where, if anywhere, this check should live for the MIR instruction.

Should this just do nothing, or `span_bug!` it like `NonDivergingIntrinsic::CopyNonOverlapping` is in this file?

Will the mir validator step have the same problem with generators?

Reply:
Turning it into span bug makes sense, we'll add logic when/if we add those rvalues before borrowck